addons/l10n_ar/models/res_partner.py

```diff
@@ -64,7 +64,7 @@ def check_vat(self):
         # NOTE by the moment we include the CUIT (VAT AR) validation also here because we extend the messages
         # errors to be more friendly to the user. In a future when Odoo improve the base_vat message errors
         # we can change this method and use the base_vat.check_vat_ar method.s
-        l10n_ar_partners = self.filtered(lambda x: x.l10n_latam_identification_type_id.l10n_ar_afip_code)
+        l10n_ar_partners = self.filtered(lambda p: bool(p.l10n_latam_identification_type_id.l10n_ar_afip_code) or p.country_code == 'AR')
```

Comment:
The lambda is expecting a boolean value.
before this change, l10n_ar_partners is always empty with the previous filter. When l10n_ar_partners is empty the `super.check_vat` will also run for the argentinian partners, and then the errors from Valentino's test will pop up. Basically we decide to handle argentinian VAT by ourselves

Reply:
In my editor it's giving squiggly lines
![image](https://github.com/odoo/odoo/assets/87525599/e17e4487-c4c6-46c2-9010-bb0b6ac42017)
But I guess it's fine in this case because the next statement `or p.country_code == 'AR'` will evaluate them to boolean. I'll remove the bool